be/src/vec/columns/column_array.cpp

```diff
@@ -119,7 +119,12 @@ ColumnArray::ColumnArray(MutableColumnPtr&& nested_column) : data(std::move(nest
 }
 
 MutableColumnPtr ColumnArray::get_shrinked_column() {
-    return ColumnArray::create(data->get_shrinked_column(), offsets->assume_mutable());
+    if (data->is_column_string() || data->is_column_array() || data->is_column_map() ||
```

Comment:
Not do like this. If we add another type that need shrink, we may forget to modify this place.
We could add a new virtual method in column.h    bool could_shrink()
and use it in get_shrinked_column to check if need call get_shrinked_column.